gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/model/phoenix/PhoenixServiceModelGenerator.java

```diff
@@ -58,14 +59,24 @@ public ServiceModel generateService(ApiService       service,
                                       ApiServiceConfig serviceConfig,
                                       ApiRole          role,
                                       ApiConfigList    roleConfig) {
+
     String hostname = role.getHostRef().getHostname();
-    // Phoenix Query Server does not support https
+    String sslEnabledRaw = getRoleConfigValue(roleConfig, SSL_ENABLED);
     String scheme = "http";
+
+    if (sslEnabledRaw != null) {
+      if (Boolean.parseBoolean(getRoleConfigValue(roleConfig, SSL_ENABLED))) {
```

Comment:
If you are going to keep `sslEnabledRaw` - then no need to `null` check here and should use `sslEnabledRaw` inside the `parseBoolean`.